modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/resource/v1_0/CategoryResourceImpl.java

```diff
@@ -141,6 +142,55 @@ public Response patchProductIdCategory(Long id, Category[] categories)
 		return responseBuilder.build();
 	}
 
+	@Override
+	public Category postProductByExternalReferenceCodeCategory(
+		String externalReferenceCode, Category category)
+		throws Exception {
+
+		CPDefinition cpDefinition =
+			_cpDefinitionService.
+				fetchCPDefinitionByCProductExternalReferenceCode(
+					externalReferenceCode, contextCompany.getCompanyId());
+
+		if (cpDefinition == null) {
+			throw new NoSuchCPDefinitionException(
+				"Unable to find product with external reference code " +
+				externalReferenceCode);
+		}
+
+		AssetCategory assetCategory = _assetCategoryService.fetchCategory(
+			category.getId());
+
+		if (assetCategory == null) {
+			throw new NoSuchCategoryException(
+				"Unable to find Category with ID: " + category.getId());
+		}
+
+		List<AssetCategory> categories =
+			_assetCategoryService.getCategories(
+				cpDefinition.getModelClassName(),
+				cpDefinition.getCPDefinitionId());
+
+		categories.add(assetCategory);
+
+		long[] assetCategoryIds = new long[0];
+
+		for (AssetCategory forCategory : categories) {
+			assetCategoryIds = ArrayUtil.append(
+				assetCategoryIds, forCategory.getCategoryId());
+		}
+
+		ServiceContext serviceContext = _serviceContextHelper.getServiceContext(
+			cpDefinition.getGroupId());
+		serviceContext.setAssetCategoryIds(assetCategoryIds);
+
+		_cpDefinitionService.updateCPDefinitionCategorization(
+			cpDefinition.getCPDefinitionId(), serviceContext);
+
+		return _categoryHelper.toProductCategory(
+			assetCategory, contextAcceptLanguage.getPreferredLocale());
+	}
+
 	private void _updateProductCategories(
```

Comment:
@victorcdp , we can just use the existing method no? You are essentially duplicating the implementation from _updateProductCategories.

Reply:
You're right, i just realized that the ending of this method is exactly the same as the same as _updateProductCategories, i'll try to fix that. Btw, about the need to get the current categories before updating, is there any better way to do that or do we leave it like that?

Reply:
i guess one problem is that the _assetCategoryService.getCategories method returns a List<AssetCategories>, and the _updateProductCategories method asks for an array of Categories as a parameter

Reply:
Your parameter should already be an array @victorcdp in the post method.

Reply:
but even if the parameter is an array of Categories, the return from the _assetCategoryService.getCategories method is a List<AssetCategories>, so i can't join them and send to the _updateProductCategories method

Reply:
_updateProductCategories already handles that using array + append, so you just need to pass in the [] categories to the existing method.

Reply:
but then the existing categories will be erased, because _updateProductCategories takes the parameter of categories[] and uses their ids as the argument for serviceContext.setAssetCategoryIds, overwriting the existing categories.